crates/bevy_render/src/primitives/mod.rs

```diff
@@ -81,11 +93,28 @@ impl Sphere {
     }
 }
 
-/// A bisecting plane that partitions 3D space into two regions.
+/// A region of 3D space, specifically a closed set whose border is a bisecting 2D plane.
+/// This bisecting plane partitions 3D space into two infinite regions,
+/// the half-space is one of those regions and includes the bisecting plane.
+///
+/// Each instance of this type is characterized by:
+/// - the bisecting plane's unit normal, normalized and pointing "inside" the half-space,
+/// - the signed distance from the bisecting plane to the origin of 3D space along the normal.
+///
+/// The distance can also be seen as:
+/// - the distance from the origin of 3D space to the bisecting plane along the inverse of the normal,
+/// - the opposite of the distance from the origin of 3D space to the bisecting plane along the normal.
+///
+/// Any point `p` is considered to be within the `HalfSpace` when the length of the projection
+/// of p on the normal is greater or equal than the opposite of the distance,
+/// meaning: if the equation `normal.dot(p) + distance >= 0.` is satisfied.
 ///
-/// Each instance of this type is characterized by the bisecting plane's unit normal and distance from the origin along the normal.
-/// Any point `p` is considered to be within the `HalfSpace` when the distance is positive,
-/// meaning: if the equation `n.p + d > 0` is satisfied.
+/// For example, the half-space containing all the points with a z-coordinate lesser
+/// or equal than `8.0` would be defined by: `HalfSpace::new(Vec3::NEG_Z.extend(-8.0))`.
+/// It includes all the points from the bisecting plane towards `NEG_Z`, and the distance
+/// from the plane to the origin is `-8.0` along `NEG_Z`.
+///
+/// It is used to define a [`Frustum`].
```

Comment:
among other things

Reply:
Is it used for something else in Bevy? I didn't want to imply it's the only usage, I just wanted to say it's only used for that in Bevy.

Reply:
Changed to `It is used to define a [`Frustum`], and for light computation.` in https://github.com/bevyengine/bevy/pull/9136/commits/dee7a95f4e34e2534c35ebcbf8a292c764dd5519

Is that okay?

Reply:
Same as aabb, those are general types and component, doc shouldn't restrict how they are used.

I'm not sure I see the point of documenting that here anyway. being able to go from the frustrum doc to the half space is interesting, not the reverse in my opinion

Reply:
I think it's good to have a concrete example of what it is used for instead of just having it and not knowing how it can be used. I don't really see how my sentence implies it can't be used for something else?